srml/executive/src/lib.rs

```diff
@@ -355,6 +358,16 @@ where
 					return TransactionValidity::Invalid(ApplyError::CantPay as i8)
 				}
 
+				// pay and burn the tip if provided.
+				let tip = xt.tip().value();
+				let weight = xt.weight(encoded_len);
```

Comment:
What is weight used for?

Reply:
This is an indicator that I will make the `Weighable` trait return the priority based on an arbitrary computation.